src/controllers/controller.h

```diff
@@ -79,24 +79,16 @@ class Controller : public QObject {
 
   protected:
     template<typename SpecificMappingType>
-    std::shared_ptr<SpecificMappingType> downcastAndTakeOwnership(
-            std::shared_ptr<LegacyControllerMapping>&& pMapping) {
+    std::unique_ptr<SpecificMappingType> downcastAndClone(const LegacyControllerMapping* pMapping) {
         // When unsetting a mapping (select 'No mapping') we receive a nullptr
-        if (pMapping == nullptr) {
+        auto* pSpecifiedMapping = dynamic_cast<const SpecificMappingType*>(pMapping);
```

Comment:
```suggestion
        const auto* pSpecifiedMapping = dynamic_cast<const SpecificMappingType*>(pMapping);
```
I think this results in a linter warning down the road otherwise.